### ISSUE TITLE
Unwanted line wrap in if/else conditions without curly brackets

### ISSUE THREAD

**Issue by LilaRest** [prettier-solidity/prettier-plugin-solidity#869]
Hi ! Firstly, thanks for this amazing plugin.

Then, a thing I visually love is to write inline if/else conditions when their content is composed by a single simple line, e.g.,
```solidity
// Comment 1 and Comment 2
if (balanceOf(msg.sender) >= 1000) priority = true;

// Instead of

// Comment 1
if (balanceOf(msg.sender) >= 1000) {
    // Comment 2
    priority = true;
}
```

This plugin seems to allow this except if the `if` statement is followed by an `else if` or `else` statement that uses curly brackets. So the above example will work but this one will receive a line wrap :
```solidity
// Comment 1 and Comment 2
if (balanceOf(msg.sender) >= 1000)
    priority = true;

    // Comment 3
else {
    // ...multi-line stuff here
}
```

As you can see the `else` comment is not anymore aligned with the `else` statement, and the if statement is not inline anymore.

However, if the `if` line doesn't exceed the Prettier's `printWidth`, the expected result would be:
```solidity
// Comment 1 and Comment 2
if (balanceOf(msg.sender) >= 1000) priority = true;

// Comment 3
else {
    // ...multi-line stuff here
}
```

**Comment by fvictorio** [comment]
I see, thanks for reporting this. Fixing this kind of comment-related stuff tends to be a PITA, but I agree it's not ideal.

For the record, prettier-js does what you expect (so we should aim for the same behavior):

```js
function f() {
  // Comment 1
  if (x > 0) x--;
  // Comment 2
  else {
    x++;
  }
}
```

**Comment by LilaRest** [comment]
@fvictorio Yeah indeed I'm used to use this syntax too in JS/TS, thanks for checking.
I'm really not familiar with your codebase so it's probably not relevant, but it looks like by adding the `// Comment 3` the formatter is considering that the `if` statement is a kind of **2 lines inline** `if` statement, which shouldn't be possible if there is not wrapping.
Any inline `if` statement that doesn't wrap should be considered as one single line.

**Comment by Janther** [comment]
Hi,

Sorry for the late reply.

This is not an old idea for me. I've actually attempted twice over the years implementing all the cases for `if`/`else` statements that prettier has in JS.
both times it took me too long since, it required me to drop current cases and reimplement them using the new comment strategy, and the branch stayed too far away in the Git tree.

Will give it yet another go at these cases and see if the third time is the charmed.
